### PR TITLE
Reflect scripts as bb tasks

### PR DESCRIPTION
**Pull request by ikappaki**
Hi,

could you please review patch to reflect all `script/`s as bb tasks. This is part of #61.

I've noticed there were a couple of graalvm options missing from the windows `compile` script (`-H:IncludeResources=DEPS_CLJ_VERSION` and `--no-server`), I used the bash `compile` script as a reference.  I've also removed the command to install graalvm's native image program, because I feel this is more of a workstation setup rather than the responsibility of  the compile task ...

If this is approved, I would like to update the CI jobs next to reference the bb tasks and remove the old scripts from the codebase. I could either do this as part of this PR or raise a new one.

Thanks,

### CODE REVIEW

**Comment by borkdude** [comment]
@ikappaki I prefer to put tasks that are longer than say 3 lines, as functions in a `bb/tasks.clj` file since it's more convenient to edit code a Clojure file (since clj-kondo and clojure-lsp, etc work better there). Can we do that?

The `--no-server` argument is deprecated now I think, but I could be wrong.

**Comment by borkdude** [comment]
A small example of that here:

https://github.com/squint-cljs/squint/blob/main/bb.edn

File: bb/tasks.clj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "Compile library to standalone jar and a native executable program."
+  []
+  (let [graalvm-home (or (System/getenv "GRAALVM_HOME")
+                         (throw (Exception. "Please set GRAALVM_HOME.")))
+        java-home (str (fs/path graalvm-home "bin"))
+        deps-clj-version (slurp "resources/DEPS_CLJ_VERSION")]
+    (println "Building deps " deps-clj-version)
+    (p/shell "lein" "deps.clj" "-Spath" "-Sdeps" "{:deps {borkdude/deps.clj {:mvn/version \"0.0.1\"}}}")
+    (p/shell "lein with-profiles +native-image do clean, uberjar")
+    (let [native-image (str (fs/path graalvm-home "bin"
```
Comment (borkdude) [inline bb/tasks.clj:17]: This is no longer necessary in newer versions of bb. So if we upgrade bb in circleci and appveyor, we can simplify this code.

File: bb/tasks.clj [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                                     (if (fs/windows?) "native-image.cmd" "native-image")))]
```
Comment (borkdude) [inline bb/tasks.clj:18]: Same here

**Comment by ikappaki** [comment]
> @ikappaki I prefer to put tasks that are longer than say 3 lines, as functions in a `bb/tasks.clj` file since it's more convenient to edit code a Clojure file (since clj-kondo and clojure-lsp, etc work better there). Can we do that?

Sure, I've just move the biggest offender to `bb/tasks.cljs` (I've kept a 4 line tasks in `bb.edn` just to be alongside the other peers).

> The `--no-server` argument is deprecated now I think, but I could be wrong.

I plan to upgrade to the latest graalvm version with your blessing after the CI move to bb, so I'll make sure I'll out for dead options.

**Comment by borkdude** [comment]
Sounds good

File: bb/tasks.clj (hunk 1 above)
Comment (ikappaki) [reply bb/tasks.clj:17]: Sorry @borkdude, it is not clear to me which line(s) you are referring to (it displays from line n14 to n15). The line I don't understand its meaning is line n.15, and it appears again  at the end of the fn which I though it was only there for testing the native executable that it works. Thanks

Comment (borkdude) [reply bb/tasks.clj:17]: Ah. I mean the test if we're in Windows or not. With `p/shell` you no longer have to specify the extension in Windows.

Comment (ikappaki) [reply bb/tasks.clj:17]: > Ah. I mean the test if we're in Windows or not. With `p/shell` you no longer have to specify the extension in Windows.

Ah, this must be something very new, I seem to remember (I could be wrong) the extension was added only when the program was without a path component.

Comment (borkdude) [reply bb/tasks.clj:17]: Ah I see! Yes, you're right, I didn't see that.

Comment (ikappaki) [reply bb/tasks.clj:17]: Is there something more you wanted me to look at here?

File: bb/tasks.clj (hunk 2 above)
Comment (ikappaki) [reply bb/tasks.clj:18]: This appears to refer to exactly the same area as the previous review comment, perhaps a duplicate? :)

**Comment by borkdude** [comment]
Can you active this script in CI, so it can be tested before merging?

**Comment by ikappaki** [comment]
> Can you activate this script in CI, so it can be tested before merging?

Hi @borkdude, 

I've replaced the test/compile scripts with bb tasks in the CI scripts, and removed them from the codebase (since they are tested to work). May I leave the `script/bump_version.clj` and `script\changelog.clj` for you to test, since I'm not sure why to do it?

The next step after this PR is to consolidate the builds under one CI system. Given there could be different argument quoting behavior between the supported java version, I'm thinking of building a matrix of [macos, ubuntu, windows]x[jdk 1.8, 11, 15, 17] to test all possible outcomes. I can do this in a separate PR. Do you have a particular preference on the CI system to use? `appveyor` is very hard for me to work (it's just that you make a change to test and it is more likely to wait for several minutes before it is started). `circleci` is good, but not sure how to do matrices efficiently, I have to repeat the job definitions. `github actions` I am more proficient with and it has high grid availability, but I don't know until this will be the case. What do you think?

Thanks,

File: appveyor.yml [hunk 3]
```diff
@@ -16,9 +16,12 @@ cache:
 
 build_script:
 - cmd: >-
-    powershell -Command "(New-Object Net.WebClient).DownloadFile('https://raw.githubusercontent.com/technomancy/leiningen/stable/bin/lein.bat', 'lein.bat')"
+    choco install lein
```
Comment (borkdude) [inline appveyor.yml:19]: I'm not a fan of making changes to CI builds if they are not necessary. The powershell command worked before. Switching to choco may slow down stuff or additional stuff may be installed which causes new unforeseen issues. The reason I'm saying this is that I've used brew on macOS CI and this was very very slow.

**Comment by borkdude** [comment]
I'm only using script/bump_version.clj and script\changelog.clj locally, so it's ok to not touch those.

**Comment by borkdude** [comment]
> The next step after this PR is to consolidate the builds under one CI system.

I'm fine with _adding_ Github actions for the test matrix. If the native build is sufficiently fast, compared to appveyor, we could switch, but let's leave it in for now.

**Comment by borkdude** [comment]
Thanks!

File: appveyor.yml (hunk 3 above)
Comment (borkdude) [reply appveyor.yml]: Thank you for explaining!

**Pull request merged**